pkg/minikube/cluster/cluster.go

```diff
@@ -62,6 +64,11 @@ const (
 //INFO lvl logging is displayed due to the kubernetes api calling flag.Set("logtostderr", "true") in its init()
 //see: https://github.com/kubernetes/kubernetes/blob/master/pkg/util/logs/logs.go#L32-L34
 func init() {
+	provision.Register("minishift", &provision.RegisteredProvisioner{
```

Comment:
> So, if this code is working at all, it is by pure chance. Do we agree?

It's not pure chance, It will work for sure because selection of the provisioner is depend of the `osReleaseID`. you are right it will add to the list and the `range provisioners` will iterate it but below code snip will decide which provisioner to run. you can see here `provisioner.CompatiableWithHost()` method decide which provisioner to return and it contain very basic step `return provisioner.OsReleaseInfo.ID == provisioner.OsReleaseID` where OsReleaseInfo.ID is what passed by the provisioner and OsReleaseID is what we define and our case it's `minishift`. Which will make sure only our provisioner should run.

```
	for _, p := range provisioners {
		provisioner := p.New(d)
		provisioner.SetOsReleaseInfo(osReleaseInfo)

		if provisioner.CompatibleWithHost() {
			log.Debugf("found compatible host: %s", osReleaseInfo.ID)
			return provisioner, nil
		}
	}
```